### DISCUSSION TITLE
[Question][Hw3] What does the coverage mean?

### DISCUSSION THREAD

**Discussion by cskwak99** [prosyslab-classroom/cs348-information-security#510] [Q&A]
Hello, 
I'm having a hard time understanding the HW3.
What does the 'coverage' mean? Can you explain me the reason why example1.cov.expected value is "3,14  4,13"? 
(why no info about line 2? why column is different in line 3 and line 4 although 'int y = 10 + x', 'int z = x - 10' are both has length 14?)

Thank you.

**Comment by KihongHeo** [comment]
Hi,

coverage shows the source code information of instructions executed at runtime.

That relies on LLVM debug information attached to each LLVM instruction. 

There might not be some discrepancies between source information and LLVM information because of the following reasons

- no var declaration in LLVM
- no 1-to-1 correspondence between C and LLVM for optimizations, white space, etc
